python-package/lightgbm/basic.py

```diff
@@ -2946,22 +2946,21 @@ def update(self, train_set=None, fobj=None):
             Should accept two parameters: preds, train_data,
             and return (grad, hess).
 
-                preds : numpy 1-D array
+                preds : numpy 1-D array or numpy 2-D array (for multi-class task)
                     The predicted values.
                     Predicted values are returned before any transformation,
                     e.g. they are raw margin instead of probability of positive class for binary task.
                 train_data : Dataset
                     The training dataset.
-                grad : list, numpy 1-D array or pandas Series
+                grad : numpy 1-D array or numpy 2-D array (for multi-class task)
```

Comment:
Not sure why we should delete the `list` type as input for grad and hess here.

Reply:
And also `pandas Series`

Reply:
This is to simplify the code and the docs because otherwise it would support list, numpy 1-D array and pandas Series or list of lists, numpy 2-D array and pandas DataFrame for multiclass task. This was briefly discussed in https://github.com/microsoft/LightGBM/pull/4925#issuecomment-1040848607 but I can keep the compatibility with those inputs if you think its valuable.

Reply:
Sorry for missing out the discussion. I think it is good to explicitly allow numpy arrays for simplicity!